File: library/src/main/java/com/e16din/lightutils/LightUtils.java
package com.e16din.lightutils;

import android.content.Context;
import android.content.res.Resources;

import java.util.Locale;

/**
 * Created by e16din on 14.08.15.
 */
public final class LightUtils {

    private static volatile Context sContext;
    private static boolean sDebug;

    private LightUtils() {
    }

    /**
     * Initialize LightUtils
     *
     * @param context Application context
     * @param debug   BuildConfig.DEBUG
     */
    public static void init(final Context context, boolean debug) {
        sContext = context.getApplicationContext();
        sDebug = debug;
    }

    public static Resources getResources() {
        return sContext.getResources();
    }

    @Deprecated
    public static Locale getCurrentLocale() {
        return getResources().getConfiguration().locale;
    }

    public static boolean isDebug() {
        return sDebug;
    }

    public static Context getContext() {
        return sContext;
    }

    public static void setContext(Context context) {
        sContext = context;
    }
}

File: LightUtils/src/main/java/com/e16din/lightutils/LightUtils.kt
package com.e16din.lightutils

import android.app.Application

import com.e16din.topactivity.TopActivity

/**
 * Created by e16din on 14.08.15.
 */
object LightUtils {

    internal var isDebug: Boolean = false

    /**
     * Initialize LightUtils
     *
     * @param app   Application object
     * @param debug BuildConfig.DEBUG
     */
    fun init(app: Application, debug: Boolean = true) {
        TopActivity.init(app)
        isDebug = debug
    }
}

fun Application.initLightUtils() {
    LightUtils.init(this)
}
